### ISSUE TITLE
feat(event)!: differentiate private and unlisted visibility

### ISSUE THREAD

**Issue by dargmuesli** [maevsi/sqitch#18]
A new element `unlisted` is to be added to the [`maevsi.event_visibility` enumeration](https://github.com/maevsi/sqitch/blob/707918bae13f4299246907ce6d474607cae86fba/src/deploy/enum_event_visibility.sql).

Currently, the `private` visibility describes the `unlisted` behavior that is to be added.
`private` will get a new meaning (similar to "archived") when `unlisted` is added.

This will be a breaking change.

See https://github.com/maevsi/maevsi/issues/1668

**Comment by sthelemann** [comment]
I need more details on the semantics of public/private/unlisted.

Are the follow assumptions correct?
* Unlisted events are only visible to the organizer. Correct?
* An organizer can create guests (formerly invitations) for private or unlisted events.
* An organizer is not allowed invite to unlisted events

Questions:
* Which are the possible visibility changes? private <-> unlisted, public <-> unlisted, public <-> private? 
* We usually (or generally?) don't have guests for public events. What is supposed to happen with guests of private or unlisted events when the event turns public?
* What is supposed to happen when an invitation has already been set out to a user (account) and then the event turns unlisted? I assume that the user will no longer have access to the event via the invitation code.

**Comment by dargmuesli** [comment]
> Unlisted events are only visible to the organizer. Correct?

no, they have the exact same visibility as with `private` right now. It's just a rename of `private`.

> An organizer can create guests (formerly invitations) for private or unlisted events.

guests management is unrelated to the event visibility, An organizer can create guests for any visibility currently and in the foreseeable future.

> An organizer is not allowed invite to unlisted events

an organizer is allowed to invite to unlisted events, that's also what's currently possible with "private" events

> Which are the possible visibility changes?

any to any other

> We usually (or generally?) don't have guests for public events. What is supposed to happen with guests of private or unlisted events when the event turns public?

We do have guests for public events right now and this won't change in the foreseeable future. If that changes, it will change in the frontend first and we can draft necessary backend changes then. We don't want to add complex limitations in the backend without a serious requirement (like we have for blocking). If frontend design for adding guests for public events was removed, it would be acceptable still if a technical expert figured out how to still add guests for their public event via api calls. We'd most likely add an achievement badge for this kind of "hacking", grant it to the user and continue to go about our days as this doesn't break any functionality for anyone behaving as the frontend suggests.
So a change to the event visibility doesn't modify the event's guests in any way. Having guests who got an invitation to a – let's say unlisted – event, then turning the event private would only lead to guests facing a 403 permission denied when they tried to follow their invitation link.
When the same event turns back to `unlisted`, everything works just as it did before the changes. It just has no side effects to any other data. The policies just do their thing, that's it.

> What is supposed to happen when an invitation has already been set out to a user (account) and then the event turns unlisted? I assume that the user will no longer have access to the event via the invitation code.

see replies above
